File: Stack & Queue/*Dota2 Senate/solution.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        r, d = deque(), deque()
        n = len(senate)
        for i, c in enumerate(senate):
            if c == "D": d.append(i)
            else: r.append(i)
        while r and d:
            i1, i2 = r.popleft(), d.popleft()
            if i1 < i2: r.append(i1 + n)
            else: d.append(i2 + n)
        if r: return "Radiant"
        else: return "Dire"
        

File: Stack & Queue/*Dota2 Senate/solution.ts
function predictPartyVictory(senate: string): string {
    let r = []
    let d = []
    for(let i = 0; i < senate.length; i++) {
        if(senate[i] === "R") r.push(i)
        else d.push(i)
    }
    while(r.length && d.length) {
        let next1 = r.shift()
        let next2 = d.shift()
        if(next1 < next2) {
            r.push(next1 + senate.length)
        } else {
            d.push(next2 + senate.length)
        }
        
    }
    return r.length ? "Radiant" : "Dire"
};